sdk/program/src/clock.rs

```diff
@@ -78,52 +108,61 @@ pub const MAX_TRANSACTION_FORWARDING_DELAY_GPU: usize = 2;
 /// More delay is expected if CUDA is not enabled (as signature verification takes longer)
 pub const MAX_TRANSACTION_FORWARDING_DELAY: usize = 6;
 
-/// Slot is a unit of time given to a leader for encoding,
-///  is some some number of Ticks long.
+/// The unit of time given to a leader for encoding a block.
+///
+/// It is some some number of _ticks_ long.
 pub type Slot = u64;
 
-/// Uniquely distinguishes every version of a slot, even if the
-/// slot number is the same, i.e. duplicate slots
+/// Uniquely distinguishes every version of a slot.
+///
+/// The `BankId` is unique even if the slot number of two different slots is the
+/// same. This can happen in the case of e.g. duplicate slots.
 pub type BankId = u64;
 
-/// Epoch is a unit of time a given leader schedule is honored,
-///  some number of Slots.
+/// The unit of time a given leader schedule is honored.
+///
+/// It lasts for some number of [`Slot`]s.
 pub type Epoch = u64;
 
 pub const GENESIS_EPOCH: Epoch = 0;
 // must be sync with Account::rent_epoch::default()
 pub const INITIAL_RENT_EPOCH: Epoch = 0;
 
-/// SlotIndex is an index to the slots of a epoch
+/// An index to the slots of a epoch.
 pub type SlotIndex = u64;
 
-/// SlotCount is the number of slots in a epoch
+/// The number of slots in a epoch.
 pub type SlotCount = u64;
 
-/// UnixTimestamp is an approximate measure of real-world time,
-/// expressed as Unix time (ie. seconds since the Unix epoch)
+/// An approximate measure of real-world time.
+///
+/// Expressed as Unix time (i.e. seconds since the Unix epoch).
 pub type UnixTimestamp = i64;
 
-/// Clock represents network time.  Members of Clock start from 0 upon
-///  network boot.  The best way to map Clock to wallclock time is to use
-///  current Slot, as Epochs vary in duration (they start short and grow
-///  as the network progresses).
+/// A representation of network time.
 ///
+/// All members of `Clock` start from 0 upon network boot.
 #[repr(C)]
 #[derive(Serialize, Deserialize, Debug, Default, PartialEq, Eq)]
 pub struct Clock {
-    /// the current network/bank Slot
+    /// The current network/bank `Slot`.
```

Comment:
This predates your work, but it feels weird to me that these are defined in relation to a Bank, but this struct is itself a member of a Bank. Wdyt about removing references to bank here?

```suggestion
    /// The current `Slot`.
```